crates/nu-parser/src/parser.rs

```diff
@@ -3684,19 +3692,56 @@ pub fn parse_list_expression(
 
             if let LiteElement::Command(_, command) = arg {
                 while spans_idx < command.parts.len() {
-                    let arg = parse_multispan_value(
-                        working_set,
-                        &command.parts,
-                        &mut spans_idx,
-                        element_shape,
-                    );
+                    let curr_span = command.parts[spans_idx];
+                    let curr_tok = working_set.get_span_contents(curr_span);
+                    let (arg, ty) = if curr_tok == b"..." {
+                        // Parse the spread operator
+                        spans_idx += 1;
+                        if spans_idx < command.parts.len() {
+                            let spread_arg = parse_multispan_value(
+                                working_set,
+                                &command.parts,
+                                &mut spans_idx,
+                                &SyntaxShape::List(Box::new(element_shape.clone())),
+                            );
+                            let elem_ty = match &spread_arg.ty {
+                                Type::List(elem_ty) => *elem_ty.clone(),
+                                _ => Type::Any,
+                            };
+                            let span = Span::new(curr_span.start, spread_arg.span.end);
+                            // TODO should this be None?
+                            let custom_completion = spread_arg.custom_completion;
```

Comment:
@fdncred Any idea if the spread operator expression needs to copy its argument's `custom_completion` field? Or can it be `None`, since the argument's own `custom_completion` field will be used?

Reply:
I have no idea what this custom_completion thing is for, other than custom completions. :) @kubouch @sholderbach @sophiajt any advice you can share here?

Reply:
I ended up going with `custom_completion: None` because `Expr::UnaryNot` also uses `None` (but confirmation that that's the right thing to do would still be great)